replication.go

```diff
@@ -1605,12 +1609,12 @@ func (cm *ContentManager) checkDeal(ctx context.Context, d *contentDeal) (int, e
 
 	if provds.DealID != 0 {
 		deal, err := cm.Api.StateMarketStorageDeal(ctx, provds.DealID, types.EmptyTSK)
-		if err != nil {
+		if err != nil || deal == nil {
 			return DEAL_CHECK_UNKNOWN, fmt.Errorf("failed to lookup deal on chain: %w", err)
 		}
 
 		pcr, err := cm.lookupPieceCommRecord(content.Cid.CID)
-		if err != nil {
+		if err != nil || pcr == nil {
```

Comment:
I added the check to `pcr` here cause the `lookupPieceCommRecord` can return `nil` pcr record which will cause the dereferencing exception.